wrappers/core18_test.go

```diff
@@ -227,6 +228,37 @@ func (s *servicesTestSuite) TestAddSnapServicesForSnapdOnClassic(c *C) {
 	c.Check(s.sysdLog, IsNil)
 }
 
+func (s *servicesTestSuite) TestAddSessionServicesWithReadOnlyFilesystem(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	info := makeMockSnapdSnap(c)
+	c.Assert(os.Remove(dirs.SnapDBusSessionPolicyDir), IsNil)
+	// XXX: ugly, is there a better way to provoke syscall.EROFS (read-only fs)?
+	c.Assert(os.Symlink("/sys/fs/cgroup", dirs.SnapDBusSessionPolicyDir), IsNil)
```

Comment:
Wait, how does that help? It *is* possible to create directories there, surely as root.

My advice would be to either mock the whole syncdir interaction or use an integration test. This has the weird mixture of the two and I'm not sure it is correct.

Reply:
Mocked and added a slightly hackish spread test.